rules/ruleset.go

```diff
@@ -754,19 +754,38 @@ func (rs ruleSet) Schema() (*schema.RuleSet, error) {
 	return res, nil
 }
 
-func (rs ruleSet) Clone() (MutableRuleSet, error) {
-	// TODO(dgromov): Do an actual deep copy that doesn't rely on .Schema().
-	schema, err := rs.Schema()
-	if err != nil {
-		return nil, err
-	}
+func (rs ruleSet) Clone() MutableRuleSet {
```

Comment:
I'd make `Schema` and `Clone` take a pointer receiver and move it up to where the other methods are defined for `ruleSet`. 

I'd also move the definition of the `MutableRuleSet` interface up to be under `RuleSet`.